DependencyInjectionContainer/ServicesDisposer.cs

```diff
@@ -7,17 +7,22 @@ internal sealed class ServicesDisposer : IDisposable
 {
     private readonly List<object> instances = new();
 
-    public void Dispose()
+    public async void Dispose()
```

Comment:
Do not use async void, especially as dispose method. Check best practices here https://learn.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-disposeasync